packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,134 @@
+      let key = option?.key;
+      let updatedUserText = '';
+      if (value) {
+        if (allowFreeform && !option && !onFormatDate) {
+          // Validate only if user did not add onFormatDate
+          validateUserInput(value, showSeconds, useHour12);
+        }
+        updatedUserText = value;
+      } else if (option) {
+        updatedUserText = option.text;
+        setErrorMessage('');
+      }
+      setUserText(updatedUserText);
+      setSelectedKey(key);
+    },
+    [allowFreeform],
+  );
+
+  const evaluatePressedKey = (event: React.KeyboardEvent<IComboBox>) => {
+    if (
+      !onFormatDate &&
+      // Only permit input of digits, space, colon, A/P/M characters
+      !(
+        (event.charCode >= KEYCHAR_0 && event.charCode <= KEYCHAR_COLON) ||
+        event.charCode === KEYCHAR_SPACE ||
+        event.charCode === KEYCHAR_A ||
+        event.charCode === KEYCHAR_M ||
+        event.charCode === KEYCHAR_P
+      )
+    ) {
+      event.preventDefault();
+    }
+  };
+
+  const validateUserInput = (userInput: string, showSeconds: boolean, useHour12: boolean) => {
+    let errorMessage = '';
+    let regexString;
+    if (useHour12) {
+      regexString = showSeconds
+        ? /((1[0-2]|0?[1-9]):([0-5][0-9]):(?:[0-5]\d) ?([AaPp][Mm]))$/
+        : /((1[0-2]|0?[1-9]):([0-5][0-9]) ?([AaPp][Mm]))$/;
+    } else {
+      regexString = showSeconds
+        ? /([0-9]|0[0-9]|1[0-9]|2[0-3]):(?:[0-5]\d):(?:[0-5]\d)$/
+        : /([0-9]|0[0-9]|1[0-9]|2[0-3]):(?:[0-5]\d)$/;
+    }
+    if (!regexString.test(userInput)) {
+      let useHour12ErrorMessage = useHour12 ? '12-hour' : '24-hour';
+      showSeconds
+        ? (errorMessage = `TimePicker format must be valid and in the ${useHour12ErrorMessage} format hh:mm:ss A.`)
+        : (errorMessage = `TimePicker format must be valid and in the ${useHour12ErrorMessage} format hh:mm A.`);
+    }
+    setErrorMessage(errorMessage);
+  };
+
+  return (
+    <div>
+      <ComboBox
+        allowFreeform={allowFreeform}
+        selectedKey={selectedKey}
+        label={label}
+        errorMessage={errorMessage}
+        options={timePickerOptions}
+        onChange={onChange}
+        text={userText}
+        onKeyPress={evaluatePressedKey}
+        {...rest}
+      />
+    </div>
+  );
+};
+
+const generateDefaultTime = (increments: number, timeRange: TimeRange) => {
+  const defaultTime = new Date();
+  if (timeRange.start >= 0) {
+    defaultTime.setHours(timeRange.start);
+  }
+  const ceiledMinute = ceilMinute(defaultTime.getMinutes(), increments);
+  if (ceiledMinute) {
+    defaultTime.setMinutes(ceiledMinute);
+  }
+  return defaultTime;
+};
+
+const getDropdownOptionsCount = (increments: number, timeRange: TimeRange) => {
+  let hoursInRange = TimeConstants.HoursInOneDay;
+  if (timeRange.start >= 0 && timeRange.end >= 0) {
+    if (timeRange.start > timeRange.end) hoursInRange = 24 - timeRange.start - timeRange.end;
```

Comment:
does this need to be `TimeConstants.HoursInOneDay` instead?

Reply:
Resolved by adding `HoursInOneDay` to `TimeConstants`.